tests/sentry/region_to_control/test_region_to_control_kafka.py

```diff
@@ -0,0 +1,81 @@
+import datetime
+import random
+
+import pytest
+from django.conf import settings
+from django.test import override_settings
+
+from sentry.models import UserIP
+from sentry.region_to_control.consumer import get_region_to_control_consumer
+from sentry.region_to_control.messages import UserIpEvent
+from sentry.region_to_control.producer import produce_user_ip
+from sentry.silo import SiloMode
+from sentry.testutils.factories import Factories
+from sentry.utils.batching_kafka_consumer import create_topics
+
+# Poll this amount of times (for 0.1 sec each) at most to wait for messages
+MAX_POLL_ITERATIONS = 100
+
+
+@pytest.fixture
+def user():
+    return Factories.create_user("admin@localhost")
+
+
+@pytest.fixture
+def random_group_id():
+    return f"test-consumer-{random.randint(0, 2 ** 16)}"
```

Comment:
random is usually a poor choice especially in tests -- it leads to inherent flakiness.  can this be a deterministic value instead?

Reply:
Sure, in this case however the randomness helps reduce false positives in the case of partitioning state influence. This follows the pattern of ingest kafka tests that generate unique group ids.  To be honest, it probably doesn't matter in this singular simple test, but in future tests it could.  My main motivation for doing so was the follow the practice of the bigger kafka test suites now ahead of future growth and changes.

Reply:
if there's false positives then I don't think randomness helps you -- it just makes them flaky false positives rather than deterministic

or perhaps I'm not understanding -- could you elaborate?

Reply:
Sure, but to be honest, given that I'm introducing just a singular test and I'm riding on a simple implementation for now, I'll accept the feedback and remove the random group id.  The main motivation for including it in the first was following the convention of other kafka consumer integration tests but in this case, I have a singular test and not much else so it isn't likely necessary.  When further work warrants it I can revisit it.